frontend/src/app/word-models/related-words/related-words.component.ts

```diff
@@ -28,15 +32,37 @@ export class RelatedWordsComponent implements OnChanges {
 
     faCheck = faCheck;
 
-    constructor(private wordModelsService: WordmodelsService) { }
+    constructor(
+        route: ActivatedRoute,
+        router: Router,
+        paramService: ParamService,
+        private wordModelsService: WordmodelsService
+    ) {
+        super(route, router, paramService);
+    }
 
     ngOnChanges(changes: SimpleChanges) {
         if (changes.corpus || changes.queryText) {
             this.getData();
         }
     }
 
+    initialize() {
+        
+    }
+
+    teardown() {
+        this.setParams({ neighbours: null });
+    }
```

Comment:
This code is repeated across many components.

I recommend adding an abstract `parameterKeys` property the parent class  (in this instance, that would be `['neighbours']`) and handling the reset action in the parent class.